velox/functions/lib/window/NthValue.cpp

```diff
@@ -265,13 +299,14 @@ class NthValueFunction : public exec::WindowFunction {
 
   const exec::WindowPartition* partition_;
 
-  // These fields are set if the offset argument is a constant value.
-  std::optional<int64_t> constantOffset_;
+  // This field is set if the offset argument is a constant value.
+  std::optional<std::variant<int32_t, int64_t>> constantOffset_;
```

Comment:
What breaks if we save int32 offset in a  `int64_t constantOffset_`.
We should be able to save int32 int it, right?

Isn't just adding signature sufficient, then?

Reply:
Thanks for the feedback, nothing would break as such because the function signature ensures the offset is of the right type in Spark/Presto. It would be better to represent an int32_t offset with the right type, instead of casting from an int64_t variable. 
What do you suggest @aditi-pandit ?

Reply:
I suggest to use int64_t and not use std::variant.

Reply:
+1. We could just use int64 for the rest of the code.

Reply:
Updated.